haystack_argilla/base.py

```diff
@@ -78,13 +137,12 @@ def on_agent_final_answer(self, final_answer):
         records=[
             {
                 "fields": {
-                    "query": query, 
-                    "retrieved_document_1": answer
+                    self.field_names[0]: query, 
```

Comment:
I would fix these things to pre-named fields instead of the using the index. I think this is done for the langchain one too

Reply:
While it was previously the case, then I thought that people can use different names for the query like query, prompt, text etc.. Also for the answer like answer, response etc. As you said, in Langchain, we have prompt and response, and these fields are validated first and throws an error if they are different than prompt and response.
What I thought here is that we can give a warning or instruction that whatever fields they have, the first one will be taken as prompt and the second will be taken as response so that we don't expect the fields to be a particular name.